userapi/internal/api.go

```diff
@@ -51,3 +56,68 @@ func (a *UserInternalAPI) QueryProfile(ctx context.Context, req *api.QueryProfil
 	res.DisplayName = prof.DisplayName
 	return nil
 }
+
+func (a *UserInternalAPI) QueryAccessToken(ctx context.Context, req *api.QueryAccessTokenRequest, res *api.QueryAccessTokenResponse) error {
+	appServiceDevice, err := a.queryAppServiceToken(ctx, req.AccessToken, req.AppServiceUserID)
```

Comment:
If `req.AppServiceUserID` is optional then please add a `req.AppServiceUserID != ""` guard around the relevant bits here.